src/Agent.Plugins/Artifact/FileContainerProvider.cs

```diff
@@ -143,11 +148,41 @@ private async Task DownloadFileContainerAsync(Guid projectId, BuildArtifact arti
                 new ExecutionDataflowBlockOptions()
                 {
                     BoundedCapacity = 5000,
-                    MaxDegreeOfParallelism = 8,
+                    MaxDegreeOfParallelism = downloadParameters.ParallelizationLimit,
                     CancellationToken = cancellationToken,
                 });
 
             await downloadBlock.SendAllAndCompleteSingleBlockNetworkAsync(fileItems, cancellationToken);
+
+            // check files (will throw an exception if a file is corrupt)
+            if (downloadParameters.CheckDownloadedFiles)
+            {
+                CheckDownloads(context, items, rootPath, containerIdAndRoot.Item2);
+            }
+        }
+
+        private void CheckDownloads(AgentTaskPluginExecutionContext context, IEnumerable<FileContainerItem> items, string rootPath, string artifactName)
```

Comment:
was this an existing option taken from typescript?

Reply:
Yes, this was an existing feature that I ported over